test_batch_sim is flaky with Scheduler

Some failues have been observed, like 
https://github.com/equinor/ert/actions/runs/7384878270/job/20088617825

test_simulation_context is also flaky with the same error message:

```
FAILED tests/unit_tests/simulator/test_batch_sim.py::test_batch_ctx_status_failing_jobs[True] - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/unit_tests/simulator/test_simulation_context.py::test_simulation_context[True] - RuntimeError: There is no current event loop in thread 'MainThread'.
= 2 failed, 1567 passed, 25 skipped, 1 xfailed, 702 warnings in 251.29s (0:04:11) =
```

and the latter executes quicker for debugging:

`pytest "tests/unit_tests/simulator/test_simulation_context.py::test_simulation_context" -sxv --count=100 -k "[True"`

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,3 +1,4 @@
+import asyncio
 import fileinput
 import os
 import pkgutil
@@ -252,6 +253,14 @@ def try_queue_and_scheduler(request, monkeypatch):
     assert scheduler_mark
     if scheduler_mark.kwargs.get("skip") and should_enable_scheduler:
         pytest.skip("Skipping running test with scheduler enabled")
+    if should_enable_scheduler:
+        # Flaky - the new scheduler needs an event loop, which might not be initialized yet.
+        #  This might be a bug in python 3.8, but it does not occur locally.
+        try:
+            asyncio.get_running_loop()
+        except RuntimeError:
+            asyncio.set_event_loop(asyncio.new_event_loop())
+
     monkeypatch.setattr(
         FeatureToggling._conf["scheduler"], "is_enabled", should_enable_scheduler
     )

